cli/cmd/deploy/deploy.go

```diff
@@ -358,6 +381,192 @@ func loginWithTelemetry(ctx context.Context, ch *cmdutil.Helper, redirectURL str
+			return pollCtx.Err()
+		case <-time.After(pollInterval):
+			// Ready to check again.
+		}
+
+		// Poll for access to the Github's user account
+		pollRes, err := c.GetGithubUserStatus(ctx, &adminv1.GetGithubUserStatusRequest{})
+		if err != nil {
+			return err
+		}
+		if !pollRes.HasAccess {
+			// Sleep and poll again
+			continue
+		}
+
+		// Emit success telemetry
+		ch.Telemetry(ctx).RecordBehavioralLegacy(activity.BehavioralEventGithubConnectedSuccess)
+
+		ch.PrintfBold(`No git remote was found.
+
+Rill projects deploy continuously when you push changes to Github.
+Therefore, your project must be on Github before you deploy it to Rill.
+		`)
+
+		ch.Print(`			
+You can continue here and Rill can create a Github Repository for you or
+you can exit the command and create a repository manually.
+		
+		`)
+		if !cmdutil.ConfirmPrompt("Do you want to continue?", "", true) {
+			ch.PrintfBold(githubSetupMsg)
+			return nil
+		}
+
+		repoOwner := pollRes.Account
+		if len(pollRes.Organizations) > 0 {
+			repoOwners := []string{pollRes.Account}
+			repoOwners = append(repoOwners, pollRes.Organizations...)
+			ch.Print("\nYou also have access to organization(s)\n\n")
+			repoOwner = cmdutil.SelectPrompt("Select Github account", repoOwners, pollRes.Account)
+		}
+		// create and verify
+		githubRepository, err := createGithubRepository(ctx, ch, pollRes, localGitPath, repoOwner)
+		if err != nil {
+			return err
+		}
+
+		printer.ColorGreenBold.Printf("\nRepository %q created successfully\n\n", *githubRepository.Name)
+		ch.Print("Pushing local project to Github\n\n")
+		// init git repo
+		repo, err := git.PlainInit(localGitPath, false)
+		if err != nil {
+			if !errors.Is(err, git.ErrRepositoryAlreadyExists) {
+				return fmt.Errorf("failed to init git repo: %w", err)
+			}
+			repo, err = git.PlainOpen(localGitPath)
+			if err != nil {
+				return fmt.Errorf("failed to open git repo: %w", err)
+			}
+		}
+
+		wt, err := repo.Worktree()
+		if err != nil {
+			return fmt.Errorf("failed to get worktree: %w", err)
+		}
+
+		// git add .
+		if err := wt.AddWithOptions(&git.AddOptions{All: true}); err != nil {
+			return fmt.Errorf("failed to add files to git: %w", err)
+		}
+
+		// git commit -m
+		_, err = wt.Commit("Auto committed by Rill", &git.CommitOptions{All: true})
+		if err != nil {
+			if !errors.Is(err, git.ErrEmptyCommit) {
+				return fmt.Errorf("failed to commit files to git: %w", err)
+			}
+		}
+
+		// Create the remote
+		_, err = repo.CreateRemote(&config.RemoteConfig{Name: "origin", URLs: []string{*githubRepository.HTMLURL}})
+		if err != nil {
+			return fmt.Errorf("failed to create remote: %w", err)
+		}
+
+		// push the changes
+		if err := repo.PushContext(ctx, &git.PushOptions{Auth: &http.BasicAuth{Username: "x-access-token", Password: pollRes.AccessToken}}); err != nil {
+			return fmt.Errorf("failed to push to remote %q : %w", *githubRepository.HTMLURL, err)
+		}
+
+		ch.Print("Local changes pushed to remote\n\n")
```

Comment:
Would suggest something like `Successfully pushed your local project to Github` to mirror the earlier log, and since this is the first push, there aren't really any "changes".